### PR TITLE
Raw data features

### PR DESCRIPTION
**Pull request by chris-langfield**

### CODE REVIEW

File: src/neurodsp/raw_metrics.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    out_df = pd.concat(dfs)
+    out_df.index.rename(["snippet_id", "channel_id"], inplace=True)
+
+    return out_df
+
+
+def compute_raw_features_snippet(sr_ap, sr_lf, t0, t1, filter_ap=None, filter_lf=None):
+    """
+    Compute raw data feature table on one snippet of AP / LF data.
```
Comment (oliche) [inline src/neurodsp/raw_metrics.py:67]: Just add that the needed input is a scipy sos filter

**Review by oliche — CHANGES_REQUESTED** [review]
Hi Chris, 
It's all good to merge for this iteration but please rename the features first.
I suggest we merge right after this and we continue on the same branch to add the channels xfeats on the destripe band.

Thanks !

File: src/neurodsp/raw_metrics.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    t1s = {"ap": t1_ap, "lf": t1_lf}
+
+    data = {}
+
+    for band in ["ap", "lf"]:
+        sr = sglx[band]
+        sl = slice(int(sr.fs * t0), int(sr.fs * t1s[band]))
+        raw = sr[sl, : -sr.nsync].T
+        dc_offset = np.mean(raw, axis=1)
+        channel_labels, xfeats = detect_bad_channels(raw, **detect_kwargs[band])
```
[the remark this answers is not in the document]
Comment (chris-langfield) [reply src/neurodsp/raw_metrics.py:106]: put `_raw` at the end to preserve convention of having the band prepended (ap or lf)

File: src/neurodsp/raw_metrics.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        residual_rms = rms(butter - destriped)
+
+        data[f"{band}_dc_offset"] = dc_offset * factor
+        data[f"{band}_raw_rms"] = raw_rms * factor
+        data[f"{band}_butter_rms"] = butter_rms * factor
+        data[f"{band}_destripe_rms"] = destripe_rms * factor
+        data[f"{band}_residual_rms"] = residual_rms * factor
+        data[f"{band}_channel_labels"] = channel_labels
+        # channel detect features
+        data[f"{band}_xcor_hf"] = xfeats["xcor_hf"]
```
[the remark this answers is not in the document]
Comment (chris-langfield) [reply src/neurodsp/raw_metrics.py:122]: added these features to destripe but again put `_destripe` at the end to preserve convention

File: src/neurodsp/raw_metrics.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        raw_rms = xfeats["rms_raw"]
+        butter_rms = rms(butter)
+        destripe_rms = rms(destriped)
+        residual_rms = rms(butter - destriped)
+
+        data[f"{band}_dc_offset"] = dc_offset * factor
+        data[f"{band}_raw_rms"] = raw_rms * factor
+        data[f"{band}_butter_rms"] = butter_rms * factor
+        data[f"{band}_destripe_rms"] = destripe_rms * factor
+        data[f"{band}_residual_rms"] = residual_rms * factor
```
[the remark this answers is not in the document]
Comment (chris-langfield) [reply src/neurodsp/raw_metrics.py:119]: done

File: src/neurodsp/raw_metrics.py (hunk 1 above)
Comment (chris-langfield) [reply src/neurodsp/raw_metrics.py:67]: fixd

**Pull request merged**